contracts/src/token/WOmni.sol

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: GPL-3.0
+// Copyright (C) 2015, 2016, 2017 Dapphub
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// Based on WETH9 by Dapphub, and WETH98 by OP Labs.
+// Modified  by Omni Network
+
+pragma solidity 0.8.15;
+
+import { IWOmni } from "src/interfaces/IWOmni.sol";
+
+/**
+ * @title WOmni
+ * @notice WOmni is a modifed version of WETH98, is a version of WETH9 upgraded for Solidity 0.8.x.
+ */
+contract WOmni is IWOmni {
```

Comment:
Should probably be called WOMNI, but we (meaning me) named our token contract Omni, instead of OMNI, so just staying consistent